corehq/apps/domain/decorators.py

```diff
@@ -668,6 +679,10 @@ def _inner(request, *args, **kwargs):
 def check_domain_migration(view_func):
     def wrapped_view(request, domain, *args, **kwargs):
         if DATA_MIGRATION.enabled(domain):
+            domain_obj = Domain.get_by_name(domain)
+            if domain_obj.redirect_url:
+                return HttpResponsePermanentRedirect(domain_obj.redirect_url)
```

Comment:
just checking if we want to consider making a redirect and not a permanent redirect?
I don't know if permanent redirect is an absolute requirement.

I say so because if the migration isn't successful and the project wants to start using SaaS environment again, they won't have a way to do that then? Just good to confirm/communicate that

Reply:
I agree that we should communicate this very clearly.

They should not use this before the migration is completed, and confirmed to have been successful.

If they enable this before then, then they _will lose data_, and the point of this feature is to avoid them losing data.

Reply:
But why not make this a non-permanent redirect?

Reply:
I chose a permanent redirect because once data has been migrated to the new environment, syncing with the old environment will always result in data loss.

We could use a non-permanent redirect (HTTP 302: Found) but I'm not sure that the meaning of the code is one that we want to convey. The device should, from the moment the data migration has completed successfully, ever be syncing with the old environment again. ... Or am I missing a scenario where that is not true?

Reply:
Offlined about this. I've switched to a temporary redirect, because people need to be able to change their minds. https://github.com/dimagi/commcare-hq/pull/34461/commits/69488f09934c37513b7ec8727921ff1b4a82c959